HDR videos don't load at all on unsupported hardware

### Steps to reproduce

![Screenshot_20230502-131256](https://user-images.githubusercontent.com/94094527/235725972-adbfb958-8f38-4c60-bd68-368ac88a038a.png)

1 - Open a hdr video (like this: https://www.youtube.com/watch?v=IDDmrzzB14M&t=0) on an old and cheap smartphone that don't support hdr

2 - Open and try to play the video on libretube 

### Expected behavior

As the smartphone don't support hdr, the video should play in normal sdr like the YouTube app or website 

### Actual behavior

The video don't play at all

### LibreTube version

0.14.1

### Android version

Android 11

### Other details

Using a moto g10, an entry level phone without hdr support 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

Seems like we get the following stacktrace:
```
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617): Video codec error
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):   java.lang.IllegalStateException
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):       at android.media.MediaCodec.native_dequeueOutputBuffer(Native Method)
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):       at android.media.MediaCodec.dequeueOutputBuffer(MediaCodec.java:2630)
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):       at com.google.android.exoplayer2.mediacodec.SynchronousMediaCodecAdapter.dequeueOutputBufferIndex(SynchronousMediaCodecAdapter.java:5)
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.drainOutputBuffer(MediaCodecRenderer.java:45)
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:72)
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):       at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:1009)
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:225)
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):       at android.os.Handler.dispatchMessage(Handler.java:101)
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):       at android.os.Looper.loop(Looper.java:164)
05-02 18:44:38.081 E/MediaCodecVideoRenderer( 2617):       at android.os.HandlerThread.run(HandlerThread.java:65)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617): Playback error
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):   com.google.android.exoplayer2.ExoPlaybackException: MediaCodecVideoRenderer error, index=0, format=Format(null, null, null, video/x-vnd.on2.vp9, vp9.2, 3567436, en, [1920, 1080, 30.0], [-1, -1]), format_supported=YES
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:357)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at android.os.Handler.dispatchMessage(Handler.java:101)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at android.os.Looper.loop(Looper.java:164)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at android.os.HandlerThread.run(HandlerThread.java:65)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):   Caused by: com.google.android.exoplayer2.video.MediaCodecVideoDecoderException: Decoder failed: OMX.qcom.video.decoder.vp9
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at com.google.android.exoplayer2.video.MediaCodecVideoRenderer.createDecoderException(MediaCodecVideoRenderer.java:5)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:230)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:1009)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:225)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       ... 3 more
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):   Caused by: java.lang.IllegalStateException
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at android.media.MediaCodec.native_dequeueOutputBuffer(Native Method)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at android.media.MediaCodec.dequeueOutputBuffer(MediaCodec.java:2630)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at com.google.android.exoplayer2.mediacodec.SynchronousMediaCodecAdapter.dequeueOutputBufferIndex(SynchronousMediaCodecAdapter.java:5)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.drainOutputBuffer(MediaCodecRenderer.java:45)
05-02 18:44:38.082 E/ExoPlayerImplInternal( 2617):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:72)
```

(Tested on a Pixel 2)

--- a/app/src/main/java/com/github/libretube/helpers/DashHelper.kt
+++ b/app/src/main/java/com/github/libretube/helpers/DashHelper.kt
@@ -24,7 +24,7 @@ object DashHelper {
         val formats: MutableList<PipedStream> = mutableListOf()
     )
 
-    fun createManifest(streams: Streams): String {
+    fun createManifest(streams: Streams, supportsHdr: Boolean): String {
         val builder: DocumentBuilder = builderFactory.newDocumentBuilder()
 
         val doc = builder.newDocument()
@@ -40,17 +40,18 @@ object DashHelper {
         val adapSetInfos = ArrayList<AdapSetInfo>()
 
         val enabledVideoCodecs = PlayerHelper.enabledVideoCodecs
-        for (stream in streams.videoStreams
+        for (
+        stream in streams.videoStreams
             // used to avoid including LBRY HLS inside the streams in the manifest
             .filter { !it.format.orEmpty().contains("HLS") }
             // filter the codecs according to the user's preferences
             .filter {
-                if (enabledVideoCodecs != "all") {
-                    it.codec?.lowercase()?.startsWith(enabledVideoCodecs) ?: true
-                } else {
-                    true
-                }
-            }) {
+                enabledVideoCodecs == "all" || it.codec.orEmpty().lowercase().startsWith(
+                    enabledVideoCodecs
+                )
+            }
+            .filter { supportsHdr || !it.quality.orEmpty().uppercase().contains("HDR") }
+        ) {
             // ignore dual format streams
             if (!stream.videoOnly!!) {
                 continue

--- a/app/src/main/java/com/github/libretube/helpers/DisplayHelper.kt
+++ b/app/src/main/java/com/github/libretube/helpers/DisplayHelper.kt
@@ -1,0 +1,22 @@
+package com.github.libretube.helpers
+
+import android.content.Context
+import android.hardware.display.DisplayManager
+import android.os.Build
+import android.view.Display
+
+object DisplayHelper {
+    /**
+     * Detect whether the device supports HDR as the ExoPlayer doesn't handle it properly
+     * Returns false on and below SDK 24
+     */
+    fun supportsHdr(context: Context): Boolean {
+        val displayManager = context.getSystemService(Context.DISPLAY_SERVICE) as DisplayManager
+        val display = displayManager.getDisplay(Display.DEFAULT_DISPLAY)
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            display.hdrCapabilities.supportedHdrTypes.isNotEmpty()
+        } else {
+            false
+        }
+    }
+}

--- a/app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt
+++ b/app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt
@@ -64,15 +64,9 @@ import com.github.libretube.extensions.hideKeyboard
 import com.github.libretube.extensions.toID
 import com.github.libretube.extensions.toastFromMainDispatcher
 import com.github.libretube.extensions.updateParameters
-import com.github.libretube.helpers.BackgroundHelper
-import com.github.libretube.helpers.DashHelper
-import com.github.libretube.helpers.ImageHelper
-import com.github.libretube.helpers.NavigationHelper
-import com.github.libretube.helpers.PlayerHelper
+import com.github.libretube.helpers.*
 import com.github.libretube.helpers.PlayerHelper.checkForSegments
 import com.github.libretube.helpers.PlayerHelper.loadPlaybackParams
-import com.github.libretube.helpers.PreferenceHelper
-import com.github.libretube.helpers.ProxyHelper
 import com.github.libretube.obj.PlayerNotificationData
 import com.github.libretube.obj.ShareData
 import com.github.libretube.obj.VideoResolution
@@ -1327,7 +1321,10 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
                 val uri = streams.dash?.let { ProxyHelper.unwrapIfEnabled(it) }?.toUri().takeIf {
                     streams.livestream || streams.videoStreams.isEmpty()
                 } ?: let {
-                    val manifest = DashHelper.createManifest(streams)
+                    val manifest = DashHelper.createManifest(
+                        streams,
+                        DisplayHelper.supportsHdr(requireContext())
+                    )
 
                     // encode to base64
                     val encoded = Base64.encodeToString(manifest.toByteArray(), Base64.DEFAULT)

